src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -148,25 +158,59 @@ public override IEnumerable<StringGetter[]> GetSignatures()
             return new[] { _args.Select<UDFArg, TexlStrings.StringGetter>(key => _ => key.NameIdent.Name.Value).ToArray() };
         }
 
+        public (IntermediateNode topNode, ScopeSymbol ruleScopeSymbol) GetIRTranslator()
+        {
+            return IRTranslator.Translate(_binding);
+        }
+
+        public RecordValue GetRuntimeRecordValue(FormulaValue[] args, CancellationToken cancellationToken)
```

Comment:
>FormulaValue[] args, [](http://example.com/codeflow?start=49&length=20)

Should we verify  these are the same length (args, _args)? Probably will never fire, but just  in case?

Reply:
I can add a verification but this can decrease our test coverage.